[2025-08-13T08:23:03.605+0000] {taskinstance.py:1957} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: weather_batch_processing.run_weather_batch_script manual__2025-08-13T08:23:02.059005+00:00 [queued]>
[2025-08-13T08:23:03.629+0000] {taskinstance.py:1957} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: weather_batch_processing.run_weather_batch_script manual__2025-08-13T08:23:02.059005+00:00 [queued]>
[2025-08-13T08:23:03.630+0000] {taskinstance.py:2171} INFO - Starting attempt 1 of 2
[2025-08-13T08:23:03.676+0000] {taskinstance.py:2192} INFO - Executing <Task(SparkSubmitOperator): run_weather_batch_script> on 2025-08-13 08:23:02.059005+00:00
[2025-08-13T08:23:03.686+0000] {standard_task_runner.py:60} INFO - Started process 522 to run task
[2025-08-13T08:23:03.694+0000] {standard_task_runner.py:87} INFO - Running: ['airflow', 'tasks', 'run', 'weather_batch_processing', 'run_weather_batch_script', 'manual__2025-08-13T08:23:02.059005+00:00', '--job-id', '4', '--raw', '--subdir', 'DAGS_FOLDER/weather_batch_dag.py', '--cfg-path', '/tmp/tmphh2ixsmr']
[2025-08-13T08:23:03.701+0000] {standard_task_runner.py:88} INFO - Job 4: Subtask run_weather_batch_script
[2025-08-13T08:23:03.845+0000] {task_command.py:423} INFO - Running <TaskInstance: weather_batch_processing.run_weather_batch_script manual__2025-08-13T08:23:02.059005+00:00 [running]> on host b8abf4c23708
[2025-08-13T08:23:04.082+0000] {taskinstance.py:2481} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='your_email@example.com' AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='weather_batch_processing' AIRFLOW_CTX_TASK_ID='run_weather_batch_script' AIRFLOW_CTX_EXECUTION_DATE='2025-08-13T08:23:02.059005+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-08-13T08:23:02.059005+00:00'
[2025-08-13T08:23:04.106+0000] {base.py:83} INFO - Using connection ID 'spark_default' for task execution.
[2025-08-13T08:23:04.108+0000] {spark_submit.py:473} INFO - Spark-Submit cmd: spark-submit --master yarn --conf spark.master=local[*] --jars /opt/infra/pyspark_apps/jars/mysql-connector-j-8.3.0.jar,/opt/infra/pyspark_apps/jars/postgresql-42.7.3.jar --name weather_batch_job --verbose --queue root.default /opt/airflow/pyspark_jobs/process_weather_data.py
[2025-08-13T08:23:07.787+0000] {spark_submit.py:649} INFO - Using properties file: null
[2025-08-13T08:23:08.061+0000] {spark_submit.py:649} INFO - Exception in thread "main" org.apache.spark.SparkException: When running with master 'yarn' either HADOOP_CONF_DIR or YARN_CONF_DIR must be set in the environment.
[2025-08-13T08:23:08.062+0000] {spark_submit.py:649} INFO - at org.apache.spark.deploy.SparkSubmitArguments.error(SparkSubmitArguments.scala:650)
[2025-08-13T08:23:08.063+0000] {spark_submit.py:649} INFO - at org.apache.spark.deploy.SparkSubmitArguments.validateSubmitArguments(SparkSubmitArguments.scala:281)
[2025-08-13T08:23:08.064+0000] {spark_submit.py:649} INFO - at org.apache.spark.deploy.SparkSubmitArguments.validateArguments(SparkSubmitArguments.scala:237)
[2025-08-13T08:23:08.064+0000] {spark_submit.py:649} INFO - at org.apache.spark.deploy.SparkSubmitArguments.<init>(SparkSubmitArguments.scala:122)
[2025-08-13T08:23:08.065+0000] {spark_submit.py:649} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2$$anon$3.<init>(SparkSubmit.scala:1108)
[2025-08-13T08:23:08.065+0000] {spark_submit.py:649} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.parseArguments(SparkSubmit.scala:1108)
[2025-08-13T08:23:08.066+0000] {spark_submit.py:649} INFO - at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:86)
[2025-08-13T08:23:08.066+0000] {spark_submit.py:649} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1125)
[2025-08-13T08:23:08.067+0000] {spark_submit.py:649} INFO - at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1134)
[2025-08-13T08:23:08.068+0000] {spark_submit.py:649} INFO - at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
[2025-08-13T08:23:08.125+0000] {taskinstance.py:2699} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 575, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master yarn --conf spark.master=local[*] --jars /opt/infra/pyspark_apps/jars/mysql-connector-j-8.3.0.jar,/opt/infra/pyspark_apps/jars/postgresql-42.7.3.jar --name weather_batch_job --verbose --queue root.default /opt/airflow/pyspark_jobs/process_weather_data.py. Error code is: 1.
[2025-08-13T08:23:08.132+0000] {taskinstance.py:1138} INFO - Marking task as UP_FOR_RETRY. dag_id=weather_batch_processing, task_id=run_weather_batch_script, execution_date=20250813T082302, start_date=20250813T082303, end_date=20250813T082308
[2025-08-13T08:23:08.156+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 4 for task run_weather_batch_script (Cannot execute: spark-submit --master yarn --conf spark.master=local[*] --jars /opt/infra/pyspark_apps/jars/mysql-connector-j-8.3.0.jar,/opt/infra/pyspark_apps/jars/postgresql-42.7.3.jar --name weather_batch_job --verbose --queue root.default /opt/airflow/pyspark_jobs/process_weather_data.py. Error code is: 1.; 522)
[2025-08-13T08:23:08.184+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-08-13T08:23:08.234+0000] {taskinstance.py:3281} INFO - 0 downstream tasks scheduled from follow-on schedule check
